Tests/test_file_jpeg2k.py

```diff
@@ -174,15 +174,8 @@ def test_16bit_jp2_roundtrips(self):
     def test_unbound_local(self):
         # prepatch, a malformed jp2 file could cause an UnboundLocalError
         # exception.
-        try:
-            jp2 = Image.open('Tests/images/unbound_variable.jp2')
-            self.assertTrue(False, 'Expecting an exception')
-        except SyntaxError as err:
-            self.assertTrue(True, 'Expecting a syntax error')
-        except IOError as err:
-            self.assertTrue(True, 'Expecting an IO error')
-        except UnboundLocalError as err:
-            self.assertTrue(False, "Prepatch error")
+        with self.assertRaises(IOError):
+            Image.open('Tests/images/unbound_variable.jp2')
```

Comment:
Perhaps something like [builtins.open](https://github.com/python-pillow/Pillow/blob/5aeb0ed9727a8c51c98d11788c0135c54e88b1b7/PIL/Image.py#L2282) could still raise something, but I think your change is fine. Thanks.